lib/share_progress/facebook_variant.rb

```diff
@@ -0,0 +1,23 @@
+require 'share_progress'
+require 'share_progress/variant'
+
+module ShareProgress
+  class FacebookVariant < Variant
+    attr_accessor :facebook_title, :facebook_thumbnail, :facebook_description
+
+    def initialize(facebook_title:, facebook_description:, facebook_thumbnail:, button:)
+      @facebook_title = facebook_title
```

Comment:
The reason I had it as required is because I had trouble envisioning a workflow that would start with an empty typed variation and then moved it along. If there's a good use case for that, it makes sense, but the individual fields are required by SP, so making them not required means we've got to build in validation around the fields during serialization to make sure that the variant actually works.